test/test.make

```diff
@@ -128,12 +129,17 @@ test_e2e: start
 
 # Execute simple unit tests.
 .PHONY: run_tests
-test: run_tests
+test: run_tests run_device_manager_tests
 RUN_TESTS = TEST_WORK=$(abspath _work) \
-	$(TEST_CMD) $(shell go list $(TEST_ARGS) | sed -e 's;$(IMPORT_PATH);.;')
+	$(TEST_CMD) $(shell $(GO) list $(TEST_ARGS) | grep -v pmem-device-manager | sed -e 's;$(IMPORT_PATH);.;')
+RUN_DM_TESTS = \
+    $(TEST_CMD) ./pkg/pmem-device-manager -exec $(SUDO)
```

Comment:
This makes "make test" depend on sudo, which is not currently a requirement and also shouldn't be. Developers should be able to build and test on a machine with a user account that isn't allowed to become root.

The other issue is that the test depends on `/dev/loop-control`, right?

Let me propose a different approach to running tests that need to be run in a certain environment:
- in the test code, check those preconditions (are root, have /dev/loop-control): if not met, skip the test
- integrate the running of unit tests into our E2E testing:
  - build test binaries with `go test -c`
  - copy test binaries into the first worker
  - run them there as root

This is more complicated, but it also gives us better test coverage, because then operations that may depend on the OS kernel of the cluster node actually get tested with that kernel. We can also do unit tests that depend on actual PMEM.

Reply:
> This makes "make test" depend on sudo, which is not currently a requirement and also shouldn't be. Developers should be able to build and test on a machine with a user account that isn't allowed to become root.
> 

To avoid this root user. I initially tried ruining these tests in a container (https://github.com/intel/pmem-csi/pull/420#issuecomment-539984710).

> The other issue is that the test depends on `/dev/loop-control`, right?
> 
> Let me propose a different approach to running tests that need to be run in a certain environment:
> 
> * in the test code, check those preconditions (are root, have /dev/loop-control): if not met, skip the test
> * integrate the running of unit tests into our E2E testing:
>   
>   * build test binaries with `go test -c`
>   * copy test binaries into the first worker
>   * run them there as root
> 
> This is more complicated, but it also gives us better test coverage, because then operations that may depend on the OS kernel of the cluster node actually get tested with that kernel. We can also do unit tests that depend on actual PMEM.

Ok, Instead of running these(device manager) unit tests in full(3-node) test cluster, I would propose to run them in a single virtual machine node. I pushed changes to support this: https://github.com/intel/pmem-csi/pull/420/commits/45f21a6507f69173e57e360488fb520e2ae32939

Reply:
This single node vm installs only dependencies needed for PMEM, such as LVM2, ndctl. No other packages like Docker, Kubernetes.

Reply:
> Ok, Instead of running these(device manager) unit tests in full(3-node) test cluster, I would propose to run them in a single virtual machine node.

That's a massive change to some already pretty complex scripts. What's the advantage compared to just using the normal VM cluster?

We need to consider two different use cases:
- a developer using this while making changes to the source code
- the CI jobs

Reply:
> That's a massive change to some already pretty complex scripts.

No, it's not that massive change as it looks in the PR. Most of the changes are indentations. It's just replacing few bits in the right place in `test/start-kubernetes.sh` script.